File: test/cases/record/test_record_multiple_types.c
#include <stdio.h>
typedef struct {
    int x;
    int y;
} point;
typedef struct {
    int width;
    int height;
} rect;
point p;

rect r;

int area;


int main(void) {
    p.x = 10;
    p.y = 20;
    r.width = 30;
    r.height = 40;
    area = (r.width * r.height);
    printf("%d", p.x);
    printf("%d", p.y);
    printf("%d", area);

    return 0;
}

File: test/cases/record/test_record_multiple_types.pas
program test;
type
  point = record
    x, y: integer
  end;
  rect = record
    width, height: integer
  end;
var
  p: point;
  r: rect;
  area: integer;

begin
  p.x := 10;
  p.y := 20;

  r.width := 30;
  r.height := 40;

  area := r.width * r.height;
  write(p.x);
  write(p.y);
  write(area)
end.
